Dockerfile

```
@@ -1,40 +1,87 @@
-# Build Python app.
-FROM python:3.6-stretch
+FROM node:14 as frontend
+
+# Make build & post-install scripts behave as if we were in a CI environment (e.g. for logging verbosity purposes).
+ARG CI=true
+
+# Install front-end dependencies.
+COPY package.json package-lock.json .babelrc.js webpack.config.js ./
+RUN npm ci --no-optional --no-audit --progress=false
+
+# Compile static files
+COPY ./tbx/static_src/ ./tbx/static_src/
+RUN npm run build:prod
+
+
+# We use Debian images because they are considered more stable than the alpine
+# ones becase they use a different C compiler. Debian images also come with
+# all useful packages required for image manipulation out of the box. They
+# however weight a lot, approx. up to 1.5GiB per built image.
+FROM python:3.8-buster as backend
+
+ARG POETRY_HOME=/opt/poetry
+ARG POETRY_VERSION=1.1.1
+
+RUN useradd tbx
 
 WORKDIR /app
 
-ENV PYTHONUNBUFFERED=1 \
+# Set default environment variables. They are used at build time and runtime.
+# If you specify your own environment variables on Heroku or Dokku, they will
+# override the ones set here. The ones below serve as sane defaults only.
+#  * PATH - Make sure that Poetry is on the PATH
+#  * PYTHONUNBUFFERED - This is useful so Python does not hold any messages
+#    from being output.
+#    https://docs.python.org/3.8/using/cmdline.html#envvar-PYTHONUNBUFFERED
+#    https://docs.python.org/3.8/using/cmdline.html#cmdoption-u
+#  * PYTHONPATH - enables use of django-admin command.
+#  * DJANGO_SETTINGS_MODULE - default settings used in the container.
+#  * PORT - default port used. Please match with EXPOSE so it works on Dokku.
+#    Heroku will ignore EXPOSE and only set PORT variable. PORT variable is
+#    read/used by Gunicorn.
+#  * WEB_CONCURRENCY - number of workers used by Gunicorn. The variable is
+#    read by Gunicorn.
+#  * GUNICORN_CMD_ARGS - additional arguments to be passed to Gunicorn. This
+#    variable is read by Gunicorn
+ENV PATH=$PATH:${POETRY_HOME}/bin \
+    PYTHONUNBUFFERED=1 \
     PYTHONPATH=/app \
     DJANGO_SETTINGS_MODULE=tbx.settings.production \
     PORT=8000 \
     WEB_CONCURRENCY=3 \
-    GUNICORN_CMD_ARGS="--max-requests 1200 --access-logfile -"
+    GUNICORN_CMD_ARGS="-c gunicorn-conf.py --max-requests 1200 --access-logfile - --timeout 25"
 
+ARG BUILD_ENV
+
+# Port exposed by this container. Should default to the port used by your WSGI
+# server (Gunicorn). This is read by Dokku only. Heroku will ignore this.
 EXPOSE 8000
 
-# Install operating system dependencies.
-RUN apt-get update -y && \
-    rm -rf /var/lib/apt/lists/*
+# Install poetry using the installer (keeps Poetry's dependencies isolated from the app's)
+RUN wget https://raw.githubusercontent.com/python-poetry/poetry/${POETRY_VERSION}/get-poetry.py && \
+    echo "474997ed1c52d243d3dda6e745ffbf04ddfa68db get-poetry.py" | sha1sum -c - && \
```

@nicklee the hash needs to be `eedf0fe5a31e5bb899efa581cbe4df59af02ea5f` for `POETRY_VERSION=1.1.1`

```suggestion
    echo "eedf0fe5a31e5bb899efa581cbe4df59af02ea5f get-poetry.py" | sha1sum -c - && \
```

---

Thanks @zerolab :)!